qt/widgets/common/src/FitScriptGeneratorModel.cpp

```diff
@@ -157,148 +108,415 @@ void FitScriptGeneratorModel::addWorkspaceDomain(
+    std::string const &fullParameter, double newValue) {
+  // Deep copy so that global ties can be removed whilst in this for loop
+  auto const globalTies = m_globalTies;
+  for (auto const &globalTie : globalTies)
+    if (fullParameter == globalTie.m_tie)
+      updateParameterValueInGlobalTie(globalTie, newValue);
 }
 
-void FitScriptGeneratorModel::updateStartX(std::string const &workspaceName,
-                                           WorkspaceIndex workspaceIndex,
-                                           double startX) {
+void FitScriptGeneratorModel::updateParameterValueInGlobalTie(
+    GlobalTie const &globalTie, double newValue) {
+  if (validGlobalTie(globalTie.m_parameter, globalTie.m_tie)) {
+    auto const domainIndex = getFunctionIndexAt(globalTie.m_parameter, 0);
+    m_fitDomains[domainIndex]->setParameterValue(
+        getAdjustedFunctionIndex(globalTie.m_parameter), newValue);
+  } else {
+    clearGlobalTie(globalTie.m_parameter);
+  }
+}
+
+void FitScriptGeneratorModel::updateAttributeValue(
+    std::string const &workspaceName, WorkspaceIndex workspaceIndex,
+    std::string const &fullAttribute, IFunction::Attribute const &newValue) {
   auto const domainIndex = findDomainIndex(workspaceName, workspaceIndex);
-  m_fitDomains[domainIndex].m_startX = startX;
+  m_fitDomains[domainIndex.value]->setAttributeValue(
+      getAdjustedFunctionIndex(fullAttribute), newValue);
 }
 
-void FitScriptGeneratorModel::updateEndX(std::string const &workspaceName,
-                                         WorkspaceIndex workspaceIndex,
-                                         double endX) {
+void FitScriptGeneratorModel::updateParameterTie(
+    std::string const &workspaceName, WorkspaceIndex workspaceIndex,
+    std::string const &fullParameter, std::string const &tie) {
+  if (validTie(tie)) {
+    auto const domainIndex = findDomainIndex(workspaceName, workspaceIndex);
+    updateParameterTie(domainIndex, fullParameter, tie);
+  } else {
+    g_log.warning("Invalid tie '" + tie + "' provided.");
+  }
+}
+
+void FitScriptGeneratorModel::updateParameterTie(
+    FitDomainIndex domainIndex, std::string const &fullParameter,
+    std::string const &fullTie) {
+  checkParameterIsNotGlobal(fullParameter);
+
+  if (m_fittingMode == FittingMode::SEQUENTIAL ||
+      isSameDomain(domainIndex.value, fullTie))
+    updateLocalParameterTie(domainIndex, fullParameter, fullTie);
+  else
+    updateGlobalParameterTie(domainIndex, fullParameter, fullTie);
+}
+
+void FitScriptGeneratorModel::updateLocalParameterTie(
+    FitDomainIndex domainIndex, std::string const &fullParameter,
+    std::string const &fullTie) {
+  auto const parameter = getAdjustedFunctionIndex(fullParameter);
+  auto const tie = getLocalTie(fullTie, m_fittingMode);
+
+  if (parameter != tie && validParameter(domainIndex, fullParameter)) {
+    if (m_fitDomains[domainIndex.value]->updateParameterTie(parameter, tie))
+      clearGlobalTie(fullParameter);
```

Comment:
I wouldn't expect it to be in the global tie, or have I missed something?

Reply:
This is necessary for if a parameter is tied globally, but then its tie is updated to be a local tie instead.

Reply:
ah ok, but wouldnt we turn the global tie off first or are you doing that behind the scenes when a local tie is added?

Reply:
Yes, the global tie is essentially just replaced by a local tie within the GUI - so that should be fine